File: WebApiSpark/src/main/java/Model/Staff.java
package Model;

import Util.DatabaseValues;

public class Staff extends IncidentElement
{
    public Staff ( )
    {
        this (
                "",
                "",
                "",
                ""
        );
    }

    public Staff (
            String accountId,
            String campusId,
            String firstName,
            String lastName
    ) {
        super(
                DatabaseValues.Table.STAFF,
                new DatabaseValues.Column[]
                        {
                                DatabaseValues.Column.ACCOUNT_ID,
                                DatabaseValues.Column.CAMPUS_ID,
                                DatabaseValues.Column.FIRST_NAME,
                                DatabaseValues.Column.LAST_NAME,

                        }
        );

        updateAttributeValue(
                DatabaseValues.Column.ACCOUNT_ID,
                accountId
        );

        updateAttributeValue(
                DatabaseValues.Column.CAMPUS_ID,
                campusId
        );


        updateAttributeValue(
                DatabaseValues.Column.FIRST_NAME,
                firstName
        );

        updateAttributeValue(
                DatabaseValues.Column.LAST_NAME,
                lastName
        );
    }

    @Override
    public String toSearchString() {
        return this.getAttributeValue( DatabaseValues.Column.FIRST_NAME ) + " " +
                this.getAttributeValue( DatabaseValues.Column.LAST_NAME );
    }

    public DatabaseValues.Column getIDColumn () {
        return DatabaseValues.Column.ACCOUNT_ID ;
    }
}

File: IRSClientAngular/src/app/component/staff/staff.ts
import { Config } from '../../util/config.service';
import { IncidentElement } from '../report/incident-element';

export class Staff implements IncidentElement {
    table: string;
    attributes: StaffAttributes;

    constructor() {
        this.attributes = new StaffAttributes();
        this.table = Config.StaffTable;
    }

    toConsole(): string {
        return "hi i'm working!";
    }
    toSearchString(): string {
        return this.attributes.FIRST_NAME 
        + " " + this.attributes.LAST_NAME 
    };
}

export class StaffAttributes {
    ACCOUNT_ID: number;
    CAMPUS_ID: number;
    FIRST_NAME: string;
    LAST_NAME: string;
}